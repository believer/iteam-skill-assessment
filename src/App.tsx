import { css, Global } from '@emotion/core'
import { Router } from '@reach/router'
import * as React from 'react'
import Movie from './Movie'
import Search from './Search'
import { SearchProvider } from './SearchContext'

const styles = css`
  * {
    box-sizing: border-box;
  }

  html,
  body {
    color: #24292e;
    font-family: 'Noto Sans', serif;
    margin: 0;
  }

  a {
    text-decoration: none;
  }

  ul {
    list-style: none;
    margin: 0;
    padding: 0;
  }

  img {
    max-width: 100%;
  }
`

const App = () => {
  return (
    <SearchProvider>
      <Global styles={styles} />
      <Router>
        <Search path="/" />
        <Movie path="/movie/:id" />
      </Router>
    </SearchProvider>
  )
}

export default App
