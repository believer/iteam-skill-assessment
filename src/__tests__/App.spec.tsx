import React from 'react'
import { render } from '@testing-library/react'
import App from '../App'

test('handles app render', () => {
  const { container } = render(<App />)

  expect(container).toMatchSnapshot()
})
