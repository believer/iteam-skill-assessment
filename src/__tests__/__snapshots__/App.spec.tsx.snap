// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles app render 1`] = `
<div>
  <div
    role="group"
    style="outline: none;"
    tabindex="-1"
  >
    <div
      class="grid grid-template"
    >
      <div
        class="bg-gray-100 fixed flex justify-center p-4 z-10 transition-fast transition-ease-in-out css-19vshoa"
      >
        <input
          aria-label="Find a movie you love"
          class="font-header border-b-2 border-gray-400 focus:border-teal-400 outline-none p-3 pl-12 text-3xl bg-transparent bg-no-repeat transition-fast transition-ease-in-out css-ctpkuz"
          placeholder="Find a movie you love"
          type="text"
          value=""
        />
      </div>
      <div
        class="grid-template-center"
      >
        <div
          class="grid mt-32 md:mt-48 css-nfkm8s"
        />
      </div>
    </div>
  </div>
</div>
`;
